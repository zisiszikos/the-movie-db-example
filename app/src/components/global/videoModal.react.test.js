import React from 'react';
import VideoModal from './videoModal.react.js';

describe('VideoModal component', () => {

    it('should render properly', () => {

        const wrapper = shallow(<VideoModal />);
        expect(wrapper.is('.video-modal')).toBe(true);
        expect(wrapper.state().active).toEqual(false);
        expect(wrapper.state().videoId).toEqual(null);
        expect(wrapper).toMatchSnapshot();

        wrapper.instance().onTrailerClicked(null, {
            videoId: 1111
        });
        expect(wrapper.state().active).toEqual(true);
        expect(wrapper.state().videoId).toEqual(1111);

        wrapper.instance().closeModal(null, {
            videoId: 1111
        });
        expect(wrapper.state().active).toEqual(false);
        expect(wrapper.state().videoId).toEqual(null);

        wrapper.instance().handleVideoModalClick();
        expect(wrapper).toMatchSnapshot();

        wrapper.unmount();
    });
});
