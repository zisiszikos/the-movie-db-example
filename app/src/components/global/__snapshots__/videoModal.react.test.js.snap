// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoModal component should render properly 1`] = `
<div
  className="video-modal inactive"
  onClick={[Function]}
>
  <div
    className="inner-container"
  >
    <div
      className="iframe-container"
    >
      <i
        className="material-icons close-icon"
      >
        close
      </i>
    </div>
  </div>
</div>
`;

exports[`VideoModal component should render properly 2`] = `
<div
  className="video-modal inactive"
  onClick={[Function]}
>
  <div
    className="inner-container"
  >
    <div
      className="iframe-container"
    >
      <i
        className="material-icons close-icon"
      >
        close
      </i>
    </div>
  </div>
</div>
`;
